modules/custom/az_migration/src/Plugin/QueueWorker/DeferredMedia.php

```diff
@@ -0,0 +1,275 @@
+      ]);
+      return;
+    }
+    // Get migration components.
+    $destination = $migration->getDestinationPlugin(FALSE);
+    $process = $migration->getProcess();
+    $id_map = $migration->getIdMap();
+    $executable = new MigrateExecutable($migration);
+
+    // Get our original IDs from the map, if available.
+    $ids = $migration->getSourcePlugin()->getIds();
+    $source_ids = array_intersect_key($data, $ids);
+    $entity_ids = reset($id_map->lookupDestinationIds($source_ids));
+    if (!$entity_ids) {
+      $entity_ids = [];
+    }
+
+    // For file migrations, we should hash existing file to see if it's changed.
+    if (($destination instanceof EntityFile) && (!empty($entity_ids))) {
+      $file = $this->entityTypeManager->getStorage('file')->load(reset($entity_ids));
+      if (!empty($file)) {
+        $orig_uri = $file->getFileUri();
+        if ($orig_uri && file_exists($orig_uri)) {
+          $data['uri'] = $orig_uri;
+          $hash = Crypt::hashBase64($body);
+          $orig_hash = Crypt::hashBase64(file_get_contents($orig_uri));
+          // File is changed. Update uri.
+          if ($hash !== $orig_hash) {
+            $uri = $this->fileSystem->createFilename($data['filename'], $data['directory']);
+            $uri = $this->fileSystem->saveData($body, $uri, FileExists::Replace);
+            if ($uri !== FALSE) {
+              $data['uri'] = $uri;
+            }
+            $this->logger->notice("Updating @orig_uri to @uri because the hash changed from @orig_hash to @hash", [
+              '@orig_uri' => $orig_uri,
+              '@uri' => $uri,
+              '@orig_hash' => $orig_hash,
+              '@hash' => $hash,
+            ]);
+          }
+        }
+      }
+    }
+
+    // Process a row through the migration.
+    $row = new Row($data + $migration->getSourceConfiguration(), $ids, FALSE);
+    $executable->processRow($row, $process);
+    $destination_ids = [];
+
+    try {
+      // See if we can successfully import the row.
+      $destination_ids = $destination->import($row, $entity_ids);
+    }
+    catch (\Exception $e) {
+      // Write exceptions to the log.
+      $id_map->saveMessage($row->getSourceIdValues(), $e->getMessage());
+    }
+    if ($destination_ids) {
+      // Update the migrate map.
+      $id_map->saveIdMapping($row, $destination_ids, MigrateIdMapInterface::STATUS_NEEDS_UPDATE);
+    }
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($data) {
+    $url = $data['url'] ?? NULL;
+    $migrations = $data['migrations'] ?? [];
+    // Bail out if we weren't given a URL.
+    if (empty($url)) {
+      return;
+    }
+    try {
+      // Request the remote resource.
+      $response = $this->httpClient->get($url, [
+        // Specify an accept header to indicate what we're interested in.
+        'headers'        => ['Accept' => static::REQUEST_ACCEPT],
+      ]);
+      $this->logger->notice("DEBUG: Performing deferred fetch of %url", [
```

Comment:
Maybe we should use a different log level for these debug messages (?)
```suggestion
      $this->logger->debug("DEBUG: Performing deferred fetch of %url", [
```